src/plugins/data/public/search/aggs/buckets/lib/time_buckets/time_buckets.test.ts

```diff
@@ -0,0 +1,126 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+import moment from 'moment';
+import { coreMock } from '../../../../../../../../../src/core/public/mocks';
+
+import { TimeBuckets } from './time_buckets';
+
+describe('TimeBuckets', () => {
+  const { uiSettings } = coreMock.createSetup();
+  uiSettings.get.mockImplementation((key: string) => {
+    if (key === 'histogram:maxBars') {
```

Comment:
nit: what do you think about using `switch` operator instead of `if`?

Reply:
done